src/formats/onenote.ts

```diff
@@ -0,0 +1,489 @@
+					await this.app.vault.createBinary(attachmentPath + '/' + attachment.name, data);
+		
+					progress.reportAttachmentSuccess(attachment.name);	
+				}
+				catch (e) {
+					progress.reportFailed(attachment.name, e);
+				}
+			});
+	
+			// Clear the attachment queue after every note
+			this.attachmentQueue = [];		
+		}
+		else { }
+	}
+	
+	// Convert OneNote styled elements to valid HTML for proper htmlToMarkdown conversion
+	styledElementToHTML(pageElement: HTMLElement): HTMLElement {
+		const styledElements = pageElement.querySelectorAll('[style]');
+		
+		// For some reason cites/quotes are not converted into Markdown (possible htmlToMarkdown bug), so we do it ourselves temporarily
+		const cites = pageElement.findAll('cite');
+		cites.forEach((cite) => cite.innerHTML = '> ' + cite.innerHTML + '<br>');
+
+		// Map styles to their elements
+		const styleMap: { [key: string]: string } = {
+			'font-weight:bold': 'b',
+			'font-style:italic': 'i',
+			'text-decoration:underline': 'u',
+			'text-decoration:line-through': 's',
+			'background-color': 'mark',
+			'font-family:Consolas': 'pre',
+		};
+
+		styledElements.forEach(element => {
+			const style = element.getAttribute('style') || '';
+			const matchingStyle = Object.keys(styleMap).find(key => style.includes(key));
+	
+			if (matchingStyle) {
+				const newElementTag = styleMap[matchingStyle];
+				const newElement = document.createElement(newElementTag);
+
+				if (newElementTag === 'pre') {
+					const code = newElement.createEl('code');
+					code.textContent = element.textContent;
+				}
+
+				else newElement.innerHTML = element.innerHTML;
+				element.replaceWith(newElement);
+			}
+		});
+
+		return pageElement;
+	}
+	
+	// OneNote returns tables with an extra empty row at the top, which we need to remove
+	fixTables(pageElement: HTMLElement): HTMLElement {
+		const tables = pageElement.querySelectorAll('table');
+  
+		tables.forEach(table => {
+			if (table.rows.length > 1) table.deleteRow(0); 
+		});
+		
+		return pageElement;
+	}
+
+	async convertDrawings(element: HTMLElement, currentFile: TFile | undefined = undefined): Promise<HTMLElement> {
+		// TODO: Convert using InkML, this is a temporary notice for users to know drawings were skipped
+		const walker = document.createTreeWalker(element, NodeFilter.SHOW_COMMENT, null);
+		let hasDrawings: boolean = false;
+
+		while (walker.nextNode()) {
+		  const commentNode = walker.currentNode as Comment;
+		  if (commentNode.nodeValue?.trim() === 'InkNode is not supported') hasDrawings = true;
+		}
+
+		if (hasDrawings) {
+			const textNode = document.createTextNode('> [!caution] This page contained an drawing which was not converted. Try exporting again later, into the same output folder.');
+			// Insert the notice at the top of the page
+			element.insertBefore(textNode, element.firstChild);	
+		}
+		else {
+			for (let i = 0; i < element.children.length; i++) {
```

Comment:
Why do we need to descend into the elements children here? Won't the treeWalker retrieve all elements of concern?